src/utils/WidgetUtils.js

```diff
@@ -233,7 +233,9 @@ export default class WidgetUtils {
         };
 
         const client = MatrixClientPeg.get();
-        const userWidgets = WidgetUtils.getUserWidgets();
+        // Get the current widgets and clone them before we modify them, otherwise
+        // we'll modify the content of the old event.
+        const userWidgets = JSON.parse(JSON.stringify(WidgetUtils.getUserWidgets()));
```

Comment:
`Object.assign({}, thingToClone)` should also work and skips the overhead of JSON in the middle.

Reply:
yeah, although only shallow, although that would probably suffice in this case...